consensus/storage_fault_monitor.go

```diff
@@ -0,0 +1,89 @@
+package consensus
```

Comment:
I think this deserves its own package, say `fault` or `fault_monitor`

Reply:
I was waffling on whether it should be in its own package from a dev standpoint or whether it rightfully belongs in consensus. I could go either way though.